client/extensions/woocommerce/state/ui/shipping/classes/test/selectors.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getUiShippingClassesState,
+	getUiShippingClasses,
+	getCurrentlyOpenShippingClass,
+	isCurrentlyOpenShippingClassNew,
+	getUsedShippingClassProps,
+} from '../selectors';
+import getState from './data/get-state';
+
+const siteId = 123;
+
+describe( 'Shipping classes UI selectors', () => {
+	/**
+	 * getUiShippingClassesState
+	 */
+
+	test( '#getUiShippingClassesState', () => {
+		const state = getState();
+		const result = getUiShippingClassesState( state, siteId );
+		const expected = state.extensions.woocommerce.ui.shipping[ siteId ].classes;
+
+		expect( result ).to.equal( expected );
+	} );
+
+	/**
+	 * getUiShippingClasses
+	 */
+
+	test( '#getUiShippingClasses', () => {
+		const state = getState( {
+			created: [ 'temp-1', 'temp-2' ],
+			deleted: [ 3 ],
+			updates: [
+				{
+					id: 1,
+					name: 'Shipping class A (updated)',
+				},
+				{
+					id: 'temp-1',
+					name: 'Shipping class D (a new one)',
+				},
+				{
+					id: 'temp-2',
+					name: 'Shipping class E (a new one)',
+				},
+				{
+					id: 'temp-2',
+					name: 'Shipping class E (a really new one)',
+				},
+			],
+		} );
+
+		const result = getUiShippingClasses( state, siteId );
+
+		expect( result ).to.have.deep.members( [
+			{
+				id: 1,
+				name: 'Shipping class A (updated)',
+			},
+			{
+				id: 2,
+				name: 'Shipping class B',
+			},
+			{
+				id: 'temp-1',
+				isNew: true,
```

Comment:
Test failing due to these leftover props.

Reply:
Not anymore.